calour/transforming.py

```diff
@@ -345,3 +351,50 @@ def method(x): return x + delta
 
     exp.data = clr(centralize(method(exp.data)))
     return exp
+
+
+@Experiment._record_sig
+def subsample_count(exp, total, replace=False, inplace=False):
+    """Randomly subsample each sample to the same number of counts.
+
+    .. warning:: This function will change the data matrix in
+    ``Experiment`` object from sparse to dense. The input
+    ``Experiment`` object should not have been normalized by total sum
+    and its data should be discrete count. The samples that have few
+    total count than ``total`` will be dropped.
+
+    Parameters
+    ----------
+    total : int, optional
+        cap the tiny values and then clr transform the data.
+    replace : bool, optional
+        If True, subsample with replacement. If False (the default), subsample without replacement
+    inplace : bool, optional
+        False (default) to create a new experiment, True to do it in place
+
+    Returns
+    -------
+    ``Experiment``
+        The subsampled experiment.
+
+    See Also
+    --------
+    skbio.stats.subsample_counts
+
+    """
+    if inplace:
+        newexp = exp
+    else:
+        newexp = deepcopy(exp)
+    if newexp.sparse:
+        newexp.sparse = False
+    # subsample_counts() require int as input
+    newexp.data = newexp.data.astype(int)
+    drops = []
+    for row in range(newexp.data.shape[0]):
+        try:
+            newexp.data[row, :] = subsample_counts(newexp.data[row, :], n=total, replace=replace)
+        except ValueError:
+            drops.append(row)
```

Comment:
👍 ok that makes sense.  `apply_along_axis` probably won't work then.  Do you want to add a comment about why this occurs for future reference?